ClinGen/clincoded
# Connects to #1389. React v15.5.x warnings.

**Technical notes:**
1. In regards to the **Accessing PropTypes via the main React package is deprecated, and will be removed in  React v16.0. Use the latest available v15.x prop-types package from npm instead.** deprecation warning, a npm `prop-types` package is required.
2. In regards to the **Accessing createClass via the main React package is deprecated, and will be removed in React v16.0. Use a plain JavaScript class instead. If you're not yet ready to migrate, create-react-class v15.x is available on npm as a temporary, drop-in replacement.** deprecation warning, a npm `create-react-class` package is required.
3. Many React components are not ready to be migrated to ES6 due to their use of `mixin` that would require more time for potential HOC (higher order component) conversion and testing. As a result, the drop-in replacement of `create-react-class` is a suitable solution for the time being.
4. In the process of addressing the deprecation warnings, some of the simple `React.createClass({...})` components (that only consist of JSX rendering) were converted to the ES6 `class MyAwesomeComponent extends Component {...}` syntax. The `Component` in this syntax is a shorthand for `React.Component` as it is imported as `import React, { Componenet } from 'react'`.
5. Additionally, many of the `var MyFancyModule = require('my-fancy-module')` were converted to the ES6 `import MyFancyModule from 'my-fancy-module'` syntax.
6. Updated the `babel.js` related npm packages to their latest versions because the older versions of `babel-core` and `babel-preset-react` failed to render the `displayName` of components in the `React Developer Tools` inspector.

**Known issues:**
1. There are still many of the simple `React.createClass({...})` JSX rendering components that are not yet converted.
2. There are also other `React.createClass({...})` components that don't have `mixin` (although they have `propTypes` and `getInitialState`) and can be converted to ES6.
3. There are still other module imports that can be further simplified or converted. Given the time constraint, they can be dealt with as changes are introduced to the individual component files.
4. The **"It looks like you're using a minified copy of the development build of React. When deploying React apps to production, make sure to use the production build which skips development warnings and is faster."** warning is not addressed as part of this PR.

**Steps to test:**
1. Because the sweeping changes affect nearly all `/components/` and `/libs/` (JS) files, both of the GCI and VCI would need to be carefully tested to ensure that no existing pages/features/functionality have been broken.
2. Upon each page load or refresh, expect the deprecation warnings of `PropTypes` and `createClass` are no longer present in the browser console.
